neverbleed.c

```diff
@@ -325,10 +344,36 @@ static int expbuf_write(struct expbuf_t *buf, int fd)
     return 0;
 }
 
+/**
+ * This function waits for the provided socket to become readable, then calls `nanosleep(1)` before returning.
+ * The intention behind sleep is to provide the application to complete its event loop before the neverbleed process starts
+ * spending CPU cycles on the time-consuming RSA operation.
+ */
+static void yield_on_data(int fd)
+{
+    fd_set rfds;
+    int ret;
+    FD_ZERO(&rfds);
+    FD_SET(fd, &rfds);
+
+    while((ret = select(fd + 1, &rfds, NULL, NULL, NULL)) == -1 && (errno == EAGAIN || errno == EINTR))
+        ;
+    if (ret == -1) {
+        dief("select(2)\n");
+    } else if (ret > 0) {
+        // yield when data is available
+        struct timespec tv = { .tv_nsec = 1 };
+        if (-1 == nanosleep(&tv, NULL)) {
+            warnf("nanosleep failed/interupted");
+        }
```

Comment:
Thank you for the changes.

If we are to handle errors here, I think the correct thing to do is ignore EINTRs (because we do need to report delivery of signals), at the same time calling `dief` if `nanosleep` fails due to other reasons (because that indicates a bug in the program).

OTOH, if we do not care about the possibility of the latter, maybe these lines can be reduced to:
```suggestion
        (void)nanosleep(&tv, NULL);
```
WDYT?